src/onPreprocess.js

```diff
@@ -34,4 +70,22 @@ export default function onPreprocess() {
                 .property('checked', this.config.displayNormalRange)
                 .property('disabled', false);
     }
+
+    //only draw the chart using data from the currently selected x-axis range
+    updateXDomain(chart);
+    this.raw_data = this.super_raw_data
+        .filter(d => d[chart.config.measure_col] === chart.currentMeasure)
+        .filter(function(f) {
+            var v = chart.config.value_col;
+            return (f[v] >= chart.x_dom[0]) & (f[v] <= chart.x_dom[1]);
```

Comment:
Need to convert `f[v]` to numeric here and perhaps replace the bitwise `&` with the standard `&&`.